src/modules/todos/todos.service.ts

```diff
@@ -23,28 +25,80 @@ export class TodosService {
     }
   }
 
-  async findAllTodos(): Promise<Todo[]> {
+  async findTodos(
+    limit?: number,
+    skip?: number,
+  ): Promise<{
+    todos: Todo[];
+    count: number;
+  }> {
     this.logger.log('finding all todos...');
     try {
-      return this.todosRepository.findAllTodos();
+      if (this.isDatabaseInited) {
+        return {
+          todos: await this.todosRepository.findTodos(limit, skip),
+          count: await this.todosRepository.getTodosCount(),
+        };
+      } else {
+        const todosCount = await this.todosRepository.getTodosCount();
+        if (todosCount) {
```

Comment:
lets move this i-else statement in else to the separate function and call it in else

Reply:
moved to separate function